lazyflow/operators/ioOperators/opStreamingH5N5Reader.py

```diff
@@ -50,30 +54,59 @@ class OpStreamingHdf5Reader(Operator):
     OutputImage = OutputSlot()
 
     H5EXTS = ['.h5', '.hdf5', '.ilp']
+    N5EXTS = ['.n5']
+
+    # monkeypatch z5py to have it similar enough to be used like h5py
```

Comment:
@k-dominik, do you think this is a good place to do the monkeypatching for z5py? I could also try to move it all to __init__

Reply:
monkey patching should happen as early as possible. how about `lazyflow/__init__.py` ? Just to make sure all our submodules see the same `z5py`.
By the way, have you checked back with constantin, whether the monkey-patching is still necessary?
I've build a new version of `z5py` which is available via `ilastik-forge`...

Reply:
I checked in z5py and it is still necessary. If I recall it correctly, Constantin and I decided to leave it like that. I moved it to ```lazyflow/__init__.py```

Reply:
I think the issues were `read_direct` and `write_direct`, which, as far as I understand, reimplement `[]` with limited slicing support in `h5py`, while potentially avoiding a memory copy.
I don't support most of the more fancy slicing in `z5py`, so this functionality is basically the same as `[]`.
That being said, we could still implement this in `z5py`. You could just make a PR with the functions below @paulhfu.